// tdesign-vue-next/es/color-picker/color-picker-panel.mjs
/**
 * tdesign v1.10.6
 * (c) 2024 tdesign
 * @license MIT
 */

import { _ as _defineProperty } from '../_chunks/dep-2cd917b7.mjs';
import { defineComponent, computed, createVNode, mergeProps } from 'vue';
import props from './props.mjs';
import ColorPanel from './panel/index.mjs';
import { usePrefixClass } from '../hooks/useConfig.mjs';
import '../_chunks/dep-c343de10.mjs';
import { _ as _arrayMap } from '../_chunks/dep-b1273bbc.mjs';
import { _ as _baseIteratee } from '../_chunks/dep-77ea84d8.mjs';
import { _ as _basePickBy } from '../_chunks/dep-99f6988f.mjs';
import { _ as _getAllKeysIn } from '../_chunks/dep-bc6946b4.mjs';
import '../_chunks/dep-b36c56dd.mjs';
import '../_chunks/dep-769cdfcc.mjs';
import '../_chunks/dep-13773b3a.mjs';
import '../_chunks/dep-b250efbd.mjs';
import '../_common/js/color-picker/constants.mjs';
import './panel/header.mjs';
import '../radio/index.mjs';
import '../radio/radio.mjs';
import '../utils/helper.mjs';
import '../_chunks/dep-62613491.mjs';
import '../_chunks/dep-9f3c7a03.mjs';
import '../_chunks/dep-8c4e0f42.mjs';
import '../_chunks/dep-5b0920c8.mjs';
import '../_chunks/dep-0d26b449.mjs';
import '../_chunks/dep-860543c8.mjs';
import '../_chunks/dep-41afea2a.mjs';
import '../_chunks/dep-0c393202.mjs';
import '../_chunks/dep-be7755a9.mjs';
import '../_chunks/dep-5d6ad0f8.mjs';
import '../_chunks/dep-b9904594.mjs';
import '../_chunks/dep-5639eb7d.mjs';
import '../_chunks/dep-da8c0faf.mjs';
import '../radio/props.mjs';
import '../radio/constants.mjs';
import '../hooks/useDisabled.mjs';
import '../_chunks/dep-208ab852.mjs';
import '../hooks/useReadonly.mjs';
import '../hooks/useVModel.mjs';
import '../_chunks/dep-90feab0a.mjs';
import '../hooks/tnode.mjs';
import '../_chunks/dep-e38b198c.mjs';
import '../_chunks/dep-05274cf8.mjs';
import '../_chunks/dep-d10845aa.mjs';
import '../utils/render-tnode.mjs';
import '../_chunks/dep-fb32dbe0.mjs';
import '../_chunks/dep-eda9976d.mjs';
import '../_chunks/dep-c7111b68.mjs';
import '../_chunks/dep-aaf0a453.mjs';
import '../_chunks/dep-2bcf3734.mjs';
import '../_chunks/dep-4e9148a5.mjs';
import '../_chunks/dep-45bc846d.mjs';
import '../_chunks/dep-f03d2f0f.mjs';
import '../_chunks/dep-67e3536c.mjs';
import '../config-provider/useConfig.mjs';
import '../_chunks/dep-5a707436.mjs';
import '../_chunks/dep-28ab9e3e.mjs';
import '../_chunks/dep-a33ab1de.mjs';
import '../_chunks/dep-653fadb3.mjs';
import '../_chunks/dep-5005acfb.mjs';
import '../_chunks/dep-754933cb.mjs';
import '../_chunks/dep-6d263d27.mjs';
import '../_chunks/dep-5d631aa0.mjs';
import '../_chunks/dep-0d4a3f7e.mjs';
import '../_chunks/dep-bd1b86d0.mjs';
import '../_chunks/dep-37ea0644.mjs';
import '../_chunks/dep-1d4985f2.mjs';
import '../_chunks/dep-ace4b3db.mjs';
import '../_common/js/global-config/default-config.mjs';
import '../_common/js/global-config/locale/zh_CN.mjs';
import '../_chunks/dep-5101fc98.mjs';
import '../_chunks/dep-06adc476.mjs';
import '../_chunks/dep-eb1a861a.mjs';
import '../config-provider/type.mjs';
import '../_chunks/dep-94da3846.mjs';
import '../_chunks/dep-81a441b7.mjs';
import '../radio/group.mjs';
import '../_chunks/dep-2ebd3a6b.mjs';
import '../_chunks/dep-7baeba65.mjs';
import '../_chunks/dep-b523a7a1.mjs';
import '../_chunks/dep-5094012f.mjs';
import '../_chunks/dep-98a67207.mjs';
import '../radio/radio-group-props.mjs';
import '../radio/radio-button.mjs';
import '../radio/useKeyboard.mjs';
import '../utils/dom.mjs';
import '../utils/easing.mjs';
import '../_common/js/common.mjs';
import '../watermark/hooks.mjs';
import '../hooks/useResizeObserver.mjs';
import '../utils/withInstall.mjs';
import './style/css.mjs';
import '../radio/type.mjs';
import './hooks.mjs';
import '../hooks/index.mjs';
import '../hooks/icon.mjs';
import '../hooks/slot.mjs';
import '../hooks/useCommonClassName.mjs';
import '../hooks/useDefaultValue.mjs';
import '../hooks/useDestroyOnClose.mjs';
import '../hooks/useKeepAnimation.mjs';
import '../hooks/useRipple.mjs';
import '../utils/set-style.mjs';
import '../hooks/useVirtualScroll.mjs';
import '../hooks/useImagePreviewUrl.mjs';
import '../_common/js/upload/utils.mjs';
import '../_common/js/log/log.mjs';
import './panel/linear-gradient.mjs';
import './utils/index.mjs';
import '../_common/js/color-picker/index.mjs';
import '../_common/js/color-picker/cmyk.mjs';
import '../_common/js/color-picker/color.mjs';
import '../_chunks/dep-4a07c5ff.mjs';
import '../_chunks/dep-d6c61759.mjs';
import '../_common/js/color-picker/gradient.mjs';
import '../_common/js/color-picker/draggable.mjs';
import '../input-number/index.mjs';
import '../input-number/input-number.mjs';
import 'tdesign-icons-vue-next';
import '../button/index.mjs';
import '../button/button.mjs';
import '../loading/index.mjs';
import '../loading/loading.mjs';
import '../loading/icon/gradient.mjs';
import '../_common/js/loading/circle-adapter.mjs';
import '../_common/js/utils/set-style.mjs';
import '../_common/js/utils/helper.mjs';
import '../loading/props.mjs';
import '../hooks/useTeleport.mjs';
import '../loading/type.mjs';
import '../loading/plugin.mjs';
import '../_chunks/dep-1dda5d5d.mjs';
import '../_chunks/dep-0b5c77ed.mjs';
import '../_chunks/dep-4a6bdbd3.mjs';
import '../_chunks/dep-0df36f23.mjs';
import '../_chunks/dep-ff82207a.mjs';
import '../_chunks/dep-29715a4a.mjs';
import '../_chunks/dep-6502b585.mjs';
import '../button/props.mjs';
import '../button/type.mjs';
import '../input/index.mjs';
import '../input/input.mjs';
import '../input/props.mjs';
import '../hooks/useGlobalIcon.mjs';
import '../input/useInput.mjs';
import '../form/const.mjs';
import '../input/useLengthLimit.mjs';
import '../_common/js/log/index.mjs';
import '../input/useInputEventHandler.mjs';
import '../input/useInputWidth.mjs';
import '../input/input-group.mjs';
import '../input/input-group-props.mjs';
import '../input/type.mjs';
import '../input-number/props.mjs';
import '../input-number/hooks/useInputNumber.mjs';
import '../_common/js/input-number/number.mjs';
import '../_common/js/input-number/large-number.mjs';
import '../input-number/type.mjs';
import './panel/base-props.mjs';
import './panel/saturation.mjs';
import './panel/hue.mjs';
import './panel/slider.mjs';
import './panel/alpha.mjs';
import './panel/format/index.mjs';
import '../select/index.mjs';
import '../select/select.mjs';
import '../_chunks/dep-145a173e.mjs';
import '../_chunks/dep-e331b803.mjs';
import '../common-components/fake-arrow.mjs';
import '../select-input/index.mjs';
import '../select-input/select-input.mjs';
import '../popup/index.mjs';
import '../popup/popup.mjs';
import '@popperjs/core';
import '../popup/container.mjs';
import '../popup/props.mjs';
import '../popup/type.mjs';
import '../select-input/props.mjs';
import '../select-input/useSingle.mjs';
import '../select-input/useMultiple.mjs';
import '../tag-input/index.mjs';
import '../tag-input/tag-input.mjs';
import '../tag-input/props.mjs';
import '../tag-input/hooks/useTagScroll.mjs';
import '../tag-input/useTagList.mjs';
import '../tag/index.mjs';
import '../tag/tag.mjs';
import '../tag/props.mjs';
import '../tag/check-tag.mjs';
import '../tag/check-tag-props.mjs';
import '../tag/check-tag-group.mjs';
import '../tag/check-tag-group-props.mjs';
import '../tag/type.mjs';
import '../tag-input/hooks/useHover.mjs';
import '../tag-input/hooks/useDragSorter.mjs';
import '../tag-input/type.mjs';
import '../select-input/useOverlayInnerStyle.mjs';
import '../_chunks/dep-89404ad2.mjs';
import '../_chunks/dep-78a897b3.mjs';
import '../select-input/type.mjs';
import '../select/select-panel.mjs';
import '../_chunks/dep-d2ce4e2b.mjs';
import '../_chunks/dep-78e0bb48.mjs';
import '../select/option.mjs';
import '../select/option-props.mjs';
import '../checkbox/index.mjs';
import '../checkbox/checkbox.mjs';
import '../checkbox/props.mjs';
import '../checkbox/constants.mjs';
import '../checkbox/hooks/useCheckboxLazyLoad.mjs';
import '../_common/js/utils/observe.mjs';
import '../checkbox/hooks/useKeyboardEvent.mjs';
import '../checkbox/group.mjs';
import '../checkbox/checkbox-group-props.mjs';
import '../checkbox/type.mjs';
import '../select/helper.mjs';
import '../select/optionGroup.mjs';
import '../select/option-group-props.mjs';
import '../select/props.mjs';
import '../select/hooks/usePanelVirtualScroll.mjs';
import '../select/hooks/useSelectOptions.mjs';
import '../select/hooks/useKeyboardControl.mjs';
import '../select/type.mjs';
import './panel/format/inputs.mjs';
import './panel/format/config.mjs';
import './panel/swatches.mjs';

var arrayMap = _arrayMap,
  baseIteratee = _baseIteratee,
  basePickBy = _basePickBy,
  getAllKeysIn = _getAllKeysIn;

/**
 * Creates an object composed of the `object` properties `predicate` returns
 * truthy for. The predicate is invoked with two arguments: (value, key).
 *
 * @static
 * @memberOf _
 * @since 4.0.0
 * @category Object
 * @param {Object} object The source object.
 * @param {Function} [predicate=_.identity] The function invoked per property.
 * @returns {Object} Returns the new object.
 * @example
 *
 * var object = { 'a': 1, 'b': '2', 'c': 3 };
 *
 * _.pickBy(object, _.isNumber);
 * // => { 'a': 1, 'c': 3 }
 */
function pickBy(object, predicate) {
  if (object == null) {
    return {};
  }
  var props = arrayMap(getAllKeysIn(object), function (prop) {
    return [prop];
  });
  predicate = baseIteratee(predicate);
  return basePickBy(object, props, function (value, path) {
    return predicate(value, path[0]);
  });
}
var pickBy_1 = pickBy;

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var _ColorPickerPanel = defineComponent({
  name: "TColorPickerPanel",
  inheritAttrs: false,
  props: _objectSpread({}, props),
  setup: function setup(props2, _ref) {
    var attrs = _ref.attrs;
    var newProps = computed(function () {
      return pickBy_1(_objectSpread(_objectSpread({}, props2), attrs), function (v) {
        return v !== void 0;
      });
    });
    var prefix = usePrefixClass();
    return function () {
      return createVNode(ColorPanel, mergeProps(newProps.value, {
        "popupProps": null,
        "close-btn": false,
        "class": "".concat(prefix.value, "-is-inline")
      }), null);
    };
  }
});

export { _ColorPickerPanel as default };
